ocamltest/main.ml

```diff
@@ -88,26 +88,37 @@ let report_error loc e bt =
   print_exn loc e bt;
   "=> error in test script"
 
-type result_summary = No_failure | Some_failure | All_skipped
-let join_result summary result =
+(* A "path summary" summarizes a list of test results, on a path
+   from the root of the test tree to the current action. *)
+type path_summary =
+  | Some_failure (* one result was a failure *)
+  | Some_skip    (* we skipped before the end *)
+  | All_pass     (* all tests so far passed *)
+
+(* A "result summary" summarizes a tree of test results *)
+type result_summary =
+  | Some_success (* One full path had only Pass *)
+  | Some_failure (* One full path had a failure *)
+  | No_success (* No failure, but all paths skipped before the end *)
```

Comment:
With the description of `No_success` it feels like you could have kept the `All_skipped` name.

Reply:
Before All_skipped means that all actions (in all branches) had skipped. Now No_success that all branches had at least one skipping action. I thought it would be nice to avoid confusion by changing the name.

Reply:
An alternative presentation could be to reuse the status type  `Pass|Fail|Skip` and rename `result_join` and `summary_join` into `sequential_join` and `parallel_join`.

Reply:
I went for this approach in #13508, let me know which one you prefer.